components/dotenv.rst

```diff
@@ -62,9 +62,52 @@ The ``load()`` method never overwrites existing environment variables. Use the
     // ...
     $dotenv->overload(__DIR__.'/.env');
 
+As you're working with the ``Dotenv`` component you'll notice that you might want
+to have different files depending on the environment you're working in. Typically
+this happens for local development or Continuous Integration where you might
+want to have different files for your ``test`` and ``dev`` environments.
+
+You can use ``Dotenv::loadEnv()`` to ease this process for you::
+
+    use Symfony\Component\Dotenv\Dotenv;
+
+    $dotenv = new Dotenv();
+    $dotenv->loadEnv(__DIR__.'/.env');
+
+The ``Dotenv`` component will then look for the correct ``.env`` file to load
+in the following order::
+
+1. If ``.env`` exists, it is loaded first. In case there's no ``.env`` file but a
+   ``.env.dist``, this one will be loaded instead.
+2. If one of the previously mentioned files contains the variable ``APP_ENV``, the
+   variable is populated and used to load environment specific files hereafter. If
+   ``APP_ENV`` is not defined in either of the previously mentioned files, ``dev`` is
+   assumed for ``APP_ENV`` and populated by default.
+3. If there's an ``.env.$env.local`` file, this one is loaded. Otherwise, it falls
+   back to ``.env.$env``.
+4. If there's an ``env.local`` it's loaded last.
+
+This might look complicated at first glance but it gives you the opportunity to commit
+multiple environment specific files that can then be adjusted to your local environment
+easily. Given you commit ``.env``, ``.env.test`` and ``.env.test`` to represent different
```

Comment:
What is the difference between the last 2 filenames?

Reply:
Lol, not enough coffee. Thx, fixed :)